qmt/geometry/geo_2d_data.py

```diff
@@ -132,10 +134,69 @@ def coord_list(self, part_name: str) -> List:
         elif isinstance(part, LineString):
             return list(np.array(part.coords.xy).T)[:]
 
+    def crop(
+        self,
+        x_min: float = None,
+        x_max: float = None,
+        y_min: float = None,
+        y_max: float = None,
+    ):
+        if x_min is not None and x_max is not None and x_min > x_max:
+            raise ValueError(f"x_max ({x_max}) must be greater than x_min ({x_min})")
+        if y_min is not None and y_max is not None and y_min > y_max:
+            raise ValueError(f"y_max ({y_max}) must be greater than y_min ({y_min})")
+        cropped_geo = Geo2DData(self.lunit)
+        x_min_old, x_max_old, y_min_old, y_max_old = self.compute_bb()
+        if x_min < x_min_old:
+            raise ValueError(f"x_min ({x_min}) smaller than minimal x ({x_min_old})")
+        if x_max > x_max_old:
+            raise ValueError(f"x_max ({x_max}) greater than maximal x ({x_max_old})")
+        if y_min < y_min_old:
+            raise ValueError(f"y_min ({y_min}) smaller than minimal y ({y_min_old})")
+        if y_max > y_max_old:
+            raise ValueError(f"y_max ({y_max}) greater than maximal y ({y_max_old})")
+        x_min = x_min_old if x_min is None else x_min
+        x_max = x_max_old if x_max is None else x_max
+        y_min = y_min_old if y_min is None else y_min
+        y_max = y_max_old if y_max is None else y_max
+        crop_poly = Polygon(
+            [[x_min, y_min], [x_min, y_max], [x_max, y_max], [x_max, y_min]]
+        )
+
+        ind_count = defaultdict(int)
+
+        for name, part in self.parts.items():
+            ind_name = name.split(":")[0] if ":" in name else name
+            cropped_part = crop_poly.intersection(part)
+            if cropped_part.is_empty:
+                continue
+            if isinstance(cropped_part, (Polygon, LineString)):
+                cropped_geo.add_part(name, cropped_part)
+                ind_count[ind_name] += 1
+            elif isinstance(cropped_part, GeometryCollection):
+                for part in cropped_part:
+                    if not (isinstance(part, Polygon) or isinstance(part, LineString)):
+                        print(type(part))
```

Comment:
Is this print statement supposed to be here?

Reply:
Oops :P